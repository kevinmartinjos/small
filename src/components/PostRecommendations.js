import React from 'react';
import {Row, Col} from 'react-bootstrap';
import PostPreview from './PostPreview';

var PropTypes = React.PropTypes;

var PostRecommendations = React.createClass({
	propTypes: {
		posts: PropTypes.array.isRequired,
	},
	render() {
		return(
			<Row>
				<Col xs={12}>
					{this.props.posts.map(function(post){
						return(
						<Col xs={12} sm={12} md={4}>
							<PostPreview key={post.id} id={post.id} title={post.title} previewText={post.preview} />
						</Col>
						)
					})}
				</Col>
			</Row>
		);
	}
});

export default PostRecommendations;
