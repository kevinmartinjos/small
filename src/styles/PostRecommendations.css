.PostRecommendations .no-padding{
	padding: 0px;
}

.PostRecommendations .recommendations-title{
	color: rgba(0.8, 0.8, 0.8, 0.8);
}