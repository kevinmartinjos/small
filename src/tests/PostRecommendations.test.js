import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PostRecommendations from '../components/PostRecommendations';
import ReactTestUtils from 'react-addons-test-utils';

test('PostRecommendations should be rendered', () => {
	var posts = [{
		id: '1',
		title: 'foo',
		content: 'bar',
		recommendations: [1]
	}];

	const component = renderer.create(
		<PostRecommendations posts={posts} />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});
