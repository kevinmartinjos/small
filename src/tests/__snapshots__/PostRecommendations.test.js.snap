exports[`test PostRecommendations should be rendered 1`] = `
<div
  className="row">
  <div
    className="col-md-3 col-sm-1 col-xs-0" />
  <div
    className="col-md-6 col-sm-10 col-xs-12">
    <div
      className="PostPreview panel panel-default"
      id={undefined}>
      <div
        className="panel-body">
        <h3>
          <a
            className="title"
            onClick={[Function]}
            style={Object {}}>
            foo
          </a>
        </h3>
        <section />
      </div>
    </div>
  </div>
  <div
    className="col-md-3 col-sm-1 col-xs-0" />
</div>
`;
